src/virtualenv/activation/via_template.py

```diff
@@ -31,9 +33,13 @@ def generate(self, creator):
         return generated
 
     def replacements(self, creator, dest_folder):
+        if "mingw" or "cygwin" or "msys" in sysconfig.get_platform():
+            virtual_env = subprocess.check_output(["cygpath", str(creator.dest)]).decode().strip()
```

Comment:
Do we need to call this subprocess command? Can't we compute from within python land?

Reply:
@gaborbernat Thanks for your feedback, I was just trying to reuse the external tools that would be available to all users with cygwin and msys2 that have already solved this problem. We could do this in Python directly, I didn't realize the other platforms were going to fail tests because of this. I think the main change is going from:
"C:/msys64/home/"
to "/c/msys64/home/"
cygpath is smart and converts it to just "/", but that isn't really necessary since the full path works fine as well.
I could do this is a regex pretty easily, is that the approach you would take as well?

Reply:
If is a simple regex, then yes let's do it inline.